test/gtest/common/test.h

```diff
@@ -88,6 +90,17 @@ class test_base {
 
     virtual void test_body() = 0;
 
+    template<typename T>
+    void wait_for_value(volatile T *var, T value,
+                        double timeout = DEFAULT_TIMEOUT_SEC) const
+    {
+        ucs_time_t deadline = ucs_get_time() + (ucs_time_from_sec(timeout) *
+                                                ucs::test_time_multiplier());
+        while ((ucs_get_time() < deadline) && (*var != value)) {
+            sched_yield();
+        }
```

Comment:
We don't expect timeout to happen, so maybe add a check at the end:
`EXPECT_GT(deadline, ucs_get_time());`

Reply:
existing tests do that check already, so i'd prefer to not make such significant change in this PR